File: sort_py/insertion_sort.py
arr = [21, 75, 40, 89, 72, 2, 45, 32, 28, 7, 56, 58, 60, 64, 50]


def insertion_sort(a):
    for i in range(len(a)):
        j = i - 1
        key = a[i]
        while j >= 0 and key < a[j]:
            a[j+1] = a[j]
            j -= 1
        a[j+1] = key


if __name__ == '__main__':
    insertion_sort(arr)
    print(arr)

File: sort_js/insertion_sort.js
var swap = require('./utils').swap;

var arr = [21, 75, 40, 89, 72, 2, 45, 32, 28, 7, 56, 58, 60, 64, 50]


function insertion_sort(a) {
    for (var i = 0; i < a.length; i++) {
        var j = i - 1,
            key = a[i];
        while (j >= 0 && key < a[j]) {
            swap(a, j, j + 1);
            j--;
        }
        a[j + 1] = key;
    }
}

(function main() {
    insertion_sort(arr);
    console.log(arr);
})();